test/e2e/kind-kcp/e2e_test.go

```diff
@@ -0,0 +1,229 @@
+const namespace = "gardener"
+
+var (
+	kubeconfigKcp         = ".kcp/admin.kubeconfig"
+	kubeconfigKcpWorkload = ".kcp/workload.kubeconfig"
+	kubeconfigGardener    = "./bin/gardener/example/provider-local/seed-kind/base/kubeconfig"
+)
+
+var _ = Describe("Manager", Ordered, Label("kind-kcp"), func() {
+	// Before running the tests, set up the environment by creating the namespace,
+	// enforce the restricted security policy to the namespace, installing CRDs,
+	// and deploying the controller.
+	BeforeAll(func() {
+		By("installing APIResourceSchemas, APIExports and APIBindings in the provider workspace")
+		Eventually(func(g Gomega) {
+			cmd := exec.Command("kubectl", "apply", "-f", "schemas/gardener")
+			output, err := utils.Run(cmd)
+			_, _ = fmt.Fprintf(GinkgoWriter, "Output:\n %s", output)
+			g.Expect(err).NotTo(HaveOccurred(), "Failed to install APIResourceSchemas, APIExports")
+		}).Should(Succeed())
+
+		cmd := exec.Command("kubectl", "apply", "-f", "schemas/binding.yaml")
+		output, err := utils.Run(cmd)
+		_, _ = fmt.Fprintf(GinkgoWriter, "Output:\n %s", output)
+		Expect(err).NotTo(HaveOccurred(), "Failed to install APIBindings")
+
+		go func() {
+			defer GinkgoRecover()
+
+			By("running the controller")
+			Expect(os.Setenv("ENABLE_WEBHOOKS", "false")).To(Succeed())
+			cmd = exec.Command("go", "run", "cmd/main.go", "--kubeconfig", kubeconfigKcp, "-gardener-kubeconfig", kubeconfigGardener)
+			controllerOutput, err := utils.Run(cmd)
+			Expect(err).NotTo(HaveOccurred(), "Failed to run the controller")
+			_, _ = fmt.Fprintf(GinkgoWriter, "Controller logs:\n %s", controllerOutput)
+		}()
+	})
+
+	// After each test, check for failures and collect logs, events,
+	// and pod descriptions for debugging.
+	AfterEach(func() {
+		specReport := CurrentSpecReport()
+		if specReport.Failed() {
+			By("Fetching Kubernetes events")
+			cmd := exec.Command("kubectl", "get", "events", "-n", namespace, "--sort-by=.lastTimestamp")
+			eventsOutput, err := utils.Run(cmd)
+			if err == nil {
+				_, _ = fmt.Fprintf(GinkgoWriter, "Kubernetes events:\n%s", eventsOutput)
+			} else {
+				_, _ = fmt.Fprintf(GinkgoWriter, "Failed to get Kubernetes events: %s", err)
+			}
+		}
+	})
+
+	SetDefaultEventuallyTimeout(2 * time.Minute)
+	SetDefaultEventuallyPollingInterval(time.Second)
+
+	Context("Manager", func() {
+		BeforeAll(func() {
+			Expect(os.Setenv("KUBECONFIG", kubeconfigKcpWorkload)).To(Succeed())
+			By("Switch to kcp consuming workload workspace")
+			Eventually(func() error {
+				return utils.EnsureAndSwitchWorkspace("gardener", "test")
+			}).To(Succeed())
+
+			By("Create APIBindings in the consuming workspace")
+			cmd := exec.Command("kubectl", "apply", "-f", "schemas/binding.yaml")
+			_, err := utils.Run(cmd)
+			Expect(err).NotTo(HaveOccurred(), "Failed to create APIBindings in the consuming workspace")
+		})
+		It("should create a GardenerShootControlPlane with a hibernated Shoot", func(ctx SpecContext) {
```

Comment:
```suggestion
		It("should create a CAPI Cluster resulting in a hibernated Shoot", func(ctx SpecContext) {
```
I think we also named it `GardenerShootControlPlane` in the other test.
After we switched to "full" CAPI integration, this does not really fit anymore, right?